app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,277 @@
+
+        setSearchResults(data);
+        setLoading(false);
+    }, [serverUrl, channelId, currentTeamId]);
+
+    const search = useCallback(() => {
+        searchUsers(term);
+    }, [searchUsers, term]);
+
+    const onSearch = useCallback((text: string) => {
+        const hasText = Boolean(text);
+        if (!hasText) {
+            clearSearch();
+            return;
+        }
+
+        setTerm(text);
+        if (searchTimeoutId.current) {
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(() => {
+            searchUsers(text);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [searchUsers, clearSearch]);
+
+    const updateNavigationButtons = useCallback(async (manage: boolean) => {
+        setButtons(componentId, {
+            rightButtons: [{
+                color: theme.sidebarHeaderTextColor,
+                enabled: true,
+                id: MANAGE_BUTTON,
+                showAsAction: 'always',
+                testID: 'manage_members.button',
+                text: formatMessage(manage ? messages.button_done : messages.button_manage),
+            }],
+        });
+    }, [theme.sidebarHeaderTextColor]);
+
+    const toggleManageEnabled = useCallback(() => {
+        updateNavigationButtons(!isManageMode);
+        setIsManageMode((prev) => !prev);
+    }, [isManageMode, updateNavigationButtons]);
+
+    const handleRemoveUser = useCallback(async (userId: string) => {
+        const pIndex = profiles.findIndex((user) => user.id === userId);
+        const mIndex = channelMembers.findIndex((m) => m.user_id === userId);
+        if (pIndex !== -1) {
+            const newProfiles = [...profiles];
+            newProfiles.splice(pIndex, 1);
+            setProfiles(newProfiles);
+
+            const newMembers = [...channelMembers];
+            newMembers.splice(mIndex, 1);
+            setChannelMembers(newMembers);
+
+            await NavigationStore.waitUntilScreensIsRemoved(USER_PROFILE);
+            showRemoveChannelUserSnackbar();
+        }
+    }, [profiles, channelMembers]);
+
+    const handleUserChangeRole = useCallback(async ({userId, schemeAdmin}: {userId: string; schemeAdmin: boolean}) => {
+        const clone = channelMembers.map((m) => {
+            if (m.user_id === userId) {
+                m.scheme_admin = schemeAdmin;
+                return m;
+            }
+            return m;
+        });
+
+        setChannelMembers(clone);
+    }, [channelMembers]);
+
+    const data = useMemo(() => {
+        const isSearch = Boolean(term);
+        if (isSearch) {
+            return filterProfilesMatchingTerm(searchResults, term);
+        }
+        return profiles;
+    }, [term, searchResults, profiles]);
+
+    useNavButtonPressed(MANAGE_BUTTON, componentId, toggleManageEnabled, [toggleManageEnabled]);
+
+    useEffect(() => {
+        mounted.current = true;
```

Comment:
There are several things mixed in this effect that seems off.

First of all, the `mounte.current` should be in a effect that only gets executed in the first render.
I also feel weird that we are getting the profiles every time the permission to manage members.

Reply:
Good catch.  There is no reason to get the profiles again when the users permission to manage members changes. The list of users does not change